crates/router/src/types/storage/refund.rs

```diff
@@ -40,24 +54,36 @@ impl RefundDbExt for Refund {
                 filter = filter.filter(dsl::payment_id.eq(pid.to_owned()));
             }
             None => {
-                filter = filter.limit(limit);
+                filter = filter.limit(limit).offset(offset);
             }
         };
 
-        if let Some(created) = refund_list_details.created {
-            filter = filter.filter(dsl::created_at.eq(created));
-        }
-        if let Some(created_lt) = refund_list_details.created_lt {
-            filter = filter.filter(dsl::created_at.lt(created_lt));
+        if let Some(time_range) = refund_list_details.time_range {
+            filter = filter.filter(dsl::created_at.ge(time_range.start_time));
+
+            if let Some(end_time) = time_range.end_time {
+                filter = filter.filter(dsl::created_at.le(end_time));
+            }
         }
-        if let Some(created_gt) = refund_list_details.created_gt {
-            filter = filter.filter(dsl::created_at.gt(created_gt));
+
+        if let Some(connector) = refund_list_details.clone().connector {
+            filter = filter.filter(dsl::connector.eq_any(connector));
         }
-        if let Some(created_lte) = refund_list_details.created_lte {
-            filter = filter.filter(dsl::created_at.le(created_lte));
+
+        if let Some(filter_currency) = refund_list_details.clone().currency {
```

Comment:
```suggestion
        if let Some(filter_currency) = refund_list_details.currency {
```
is clone needed ?
you are anyways transforming the value via foreign_into